ArduPlane/quadplane.cpp

```diff
@@ -3289,7 +3289,8 @@ bool QuadPlane::do_vtol_takeoff(const AP_Mission::Mission_Command& cmd)
             return false;
         }
     } else {
-        plane.next_WP_loc.alt = plane.current_loc.alt + cmd.content.location.alt;
```

Comment:
Pre-existing, but the check on line 3287 looks wrong - I think it's possible those alts could be in different frames.

Reply:
you're right, I've pushed a fix and tested in SITL. Terrain alt VTOL takeoff did not work